### PR TITLE
[MM-16620] Properly handle analytics client life cycle

### PR DESCRIPTION
**Pull request by gabrieljackson**
This change improves the life cycle management of the analytics
client. It does so in two primary ways:
 - Move the client from being a shared pointer to belonging to the
   App struct.
 - Expose the client.Close() method for proper shutdown of the
   client.

This resolves the issue with the diagnostic test panicking due to delayed logging.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-16620

### CODE REVIEW

**Comment by gabrieljackson** [comment]
A note on this change: it currently only calls for the diagnostic client to be `Close()`ed on test teardown. Please share input on if I should include this in standard server shutdown as well.

**Comment by grundleborg** [comment]
@gabrieljackson what I'm seeing there all looks good. And I definitely think it should be included in standard server shutdown too. It would look bad if you're a user and you see the occasional panic on shutdown due to this.

File: app/app.go [hunk 1]
```diff
@@ -43,6 +44,8 @@ type App struct {
 	Metrics          einterfaces.MetricsInterface
 	Saml             einterfaces.SamlInterface
 
+	diagnosticClient *analytics.Client
```
Comment (crspeller) [inline app/app.go]: This is a server wide feature so it should probably be on the Server struct and initialized and shutdown with the rest of the stuff there.

The app only lives for each request.

Comment (gabrieljackson) [reply app/app.go]: I literally just figured that out in a quick chat with @jwilander. 

Thanks for confirming!

**Comment by gabrieljackson** [comment]
I have moved the the client to the server object and we now try to cleanly close it on server shutdown.

File: app/diagnostics.go [hunk 2]
```diff
@@ -56,7 +56,28 @@ const (
 	TRACK_PLUGINS  = "plugins"
 )
 
-var client *analytics.Client
+func (a *App) initDiagnostics(endpoint string) {
```
Comment (crspeller) [inline app/diagnostics.go]: Shouldn't we move these two to server too?

**Pull request merged**